spec.html

```diff
@@ -120,29 +120,87 @@ <h1>[[Call]] ( _thisArgument_, _argumentsList_ )</h1>
 		</emu-note>
 	</emu-clause>
-
-	<emu-clause id="sec-wrappedfunctioncreate" aoid="WrappedFunctionCreate">
-		<h1>WrappedFunctionCreate ( _callerRealm_, _targetFunction_ )</h1>
-		<p>The abstract operation WrappedFunctionCreate takes arguments _callerRealm_ and _targetFunction_. It is used to specify the creation of new wrapped function exotic objects. It performs the following steps when called:</p>
-		<emu-alg>
-			1. Assert: _callerRealm_ is a Realm Record.
-			1. Assert: IsCallable(_targetFunction_) is *true*.
-			1. Let _internalSlotsList_ be the internal slots listed in <emu-xref href="#table-internal-slots-of-wrapped-function-exotic-objects"></emu-xref>, plus [[Prototype]] and [[Extensible]].
-			1. Let _obj_ be ! MakeBasicObject(_internalSlotsList_).
-			1. Set _obj_.[[Prototype]] to _callerRealm_.[[Intrinsics]].[[%Function.prototype%]].
-			1. Set _obj_.[[Call]] as described in <emu-xref href="#sec-wrapped-function-exotic-objects-call-thisargument-argumentslist"></emu-xref>.
-			1. Set _obj_.[[WrappedTargetFunction]] to _targetFunction_.
-			1. Set _obj_.[[Realm]] to _callerRealm_.
-			1. Return _obj_.
-		</emu-alg>
-	</emu-clause>
 </emu-clause>
 
 <emu-clause id="sec-shadowrealm-objects">
 	<h1>ShadowRealm Objects</h1>
 	<emu-clause id="sec-shadowrealm-abstracts">
 		<h1>ShadowRealm Abstract Operations</h1>
 
+		<emu-clause id="sec-wrappedfunctioncreate" aoid="WrappedFunctionCreate">
+			<h1>WrappedFunctionCreate ( _callerRealm_, _Target_ )</h1>
+			<p>The abstract operation WrappedFunctionCreate takes arguments _callerRealm_ and _Target_. It is used to specify the creation of new wrapped function exotic objects. It performs the following steps when called:</p>
+			<emu-alg>
+				1. Assert: _callerRealm_ is a Realm Record.
+				1. Assert: IsCallable(_Target_) is *true*.
+				1. Let _internalSlotsList_ be the internal slots listed in <emu-xref href="#table-internal-slots-of-wrapped-function-exotic-objects"></emu-xref>, plus [[Prototype]] and [[Extensible]].
+				1. Let _wrapped_ be ! MakeBasicObject(_internalSlotsList_).
+				1. Set _wrapped_.[[Prototype]] to _callerRealm_.[[Intrinsics]].[[%Function.prototype%]].
+				1. Set _wrapped_.[[Call]] as described in <emu-xref href="#sec-wrapped-function-exotic-objects-call-thisargument-argumentslist"></emu-xref>.
+				1. Set _wrapped_.[[WrappedTargetFunction]] to _Target_.
+				1. Set _wrapped_.[[Realm]] to _callerRealm_.
+				1. Let _result_ be CopyNameAndLength(_wrapped_, _Target_, *"wrapped"*).
+				1. If _result_ is an Abrupt Completion, throw a TypeError exception.
+				1. Return _wrapped_.
+			</emu-alg>
+		</emu-clause>
+
+		<emu-clause id="sec-copynameandlength" aoid="CopyNameAndLength">
+			<h1>CopyNameAndLength ( _F_, _Target_, _prefix_ [ , _argCount_ ] )</h1>
```

Comment:
```suggestion
			<h1>
			  CopyNameAndLength (
			    _F_: a function object,
			    _Target_: a function object,
			    _prefix_: a String,
			    optional _argCount_: a Number,
			  )
			</h1>
```

Reply:
is this a new notation?

Reply:
yes, the entire spec uses it for AOs since https://github.com/tc39/ecma262/pull/545 was merged in July.